tests/fsharp/Compiler/CodeGen/LiteralValue.fs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.UnitTests.CodeGen.EmittedIL
+
+open FSharp.Compiler.UnitTests
+open NUnit.Framework
+
+[<TestFixture>]
+module ``Literal Value`` =
+
+    [<Test>]
+    let ``Literal Value``() =
+        CompilerAssert.CompileLibraryAndVerifyIL
+            """
+module LiteralValue
+
+[<Literal>]
+let x = 7
+
+[<EntryPoint>]
+let main _ =
+    0
+            """
+            (fun verifier -> verifier.VerifyIL [
+            """
+.class public abstract auto ansi sealed LiteralValue
+       extends [mscorlib]System.Object
+{
+  .custom instance void [FSharp.Core]Microsoft.FSharp.Core.CompilationMappingAttribute::.ctor(valuetype [FSharp.Core]Microsoft.FSharp.Core.SourceConstructFlags) = ( 01 00 07 00 00 00 00 00 )
+  .field public static literal int32 x = int32(0x00000007)
+  .custom instance void [FSharp.Core]Microsoft.FSharp.Core.LiteralAttribute::.ctor() = ( 01 00 00 00 )
+  .custom instance void [mscorlib]System.Diagnostics.DebuggerBrowsableAttribute::.ctor(valuetype [mscorlib]System.Diagnostics.DebuggerBrowsableState) = ( 01 00 00 00 00 00 00 00 )
+  .method public static int32  main(string[] _arg1) cil managed
+  {
+    .custom instance void [FSharp.Core]Microsoft.FSharp.Core.EntryPointAttribute::.ctor() = ( 01 00 00 00 )
+
```

Comment:
In general I do not like that we remove comments from emitted IL code 
https://github.com/dotnet/fsharp/blob/8209f2f6e297e98215a6a6c005d7a6b75d99a948/tests/fsharp/Compiler/ILChecker.fs#L46-L54
and do not do it for `expectedIL` samples.

In this test I have to remove all comments manually.
Can we apply the same transformation to emitted and expected IL? What do think @TIHan ?

Reply:
This is necessary since some tools (ILDasm?) add comments such that you cannot verify IL emitted by the compiler. This was necessary for string concat optimization work I believe.

Reply:
@cartermp I got it, but I think that would be nice to allow copy generated IL from ILDasm and use it in test directly (without manual comments removal). 

Then test framework (inside `verifier.VerifyIL`):
1. compile code sample and generate IL using `ILDASM`
1. remove comments from IL produced in step `1` (like it does now)
1. remove comments from expected IL specified in using test, the same transformation that we do in  step `2` (I propose)
1. compare two set of IL instructions from `2` & `3`

benefits: less manual work to cleanup `expected IL` (ability to use snippet from ILDasm).